src/libPMacc/PMaccConfig.cmake

```diff
@@ -167,41 +155,93 @@ if(VAMPIR_ENABLE)
         -v)
 
     # for manual instrumentation and hints that vampir is enabled in our code
-    add_definitions(-DVTRACE)
+    set(PMacc_DEFINITIONS ${PMacc_DEFINITIONS} -DVTRACE)
 
     # titan work around: currently (5.14.4) the -D defines are not provided by -vt:showme-compile
-    add_definitions(-DMPICH_IGNORE_CXX_SEEK)
+    set(PMacc_DEFINITIONS ${PMacc_DEFINITIONS} -DMPICH_IGNORE_CXX_SEEK)
 endif(VAMPIR_ENABLE)
 
+
 ################################################################################
-# MPI LIB
+# Find MPI
 ################################################################################
-find_package(MPI MODULE REQUIRED)
+
+find_package(MPI REQUIRED)
 set(PMacc_INCLUDE_DIRS ${PMacc_INCLUDE_DIRS} ${MPI_C_INCLUDE_PATH})
 set(PMacc_LIBRARIES ${PMacc_LIBRARIES} ${MPI_C_LIBRARIES})
-set(PMacc_LIBRARIES ${PMacc_LIBRARIES} ${MPI_CXX_LIBRARIES})
+
+# bullxmpi fails if it can not find its c++ counter part
```

Comment:
good one putting that in a guard, but I am not sure if `MPI_CXX_FOUND` is set to true if one is not specifically requesting that `COMPONENT` in `find_package`, but might work. (Side check in taurus would be nice.)